src/gpuarray/collectives.h

```diff
@@ -19,6 +19,17 @@ extern "C" {
 /**
  * \brief TODO
  * \param src [const GpuArray*] TODO
+ * \param opcode [int] TODO
+ * \param root [int] TODO
+ * \param comm [gpucomm*] TODO
+ * \return int TODO
+ */
+GPUARRAY_PUBLIC int GpuArray_reduce_from(const GpuArray* src, int opcode, int root,
+                                         gpucomm* comm);
```

Comment:
It's simply the same as GpuArray_reduce, but for non root ranks of communicator world. Should it go to array.h/c, array.c will include `gpuarray/buffer_collectives.h`. Is this good?